lib/board.rb

```diff
@@ -0,0 +1,52 @@
+class Board
+	attr_accessor :cells, :turns_taken
+	def initialize
+		@cells = Array.new(9, " ")
+		@turns_taken = []
+	end
+
+	def reset!
+		@cells = Array.new(9, " ")
+	end
+
+	def display
+		puts " #{cells[0]} | #{cells[1]} | #{cells[2]} ", 
+		"-----------", 
+		" #{cells[3]} | #{cells[4]} | #{cells[5]} ", 
+		"-----------", 
+		" #{cells[6]} | #{cells[7]} | #{cells[8]} "
+	end
+
+	def position(input)
+		return self.cells[input.to_i - 1]
+	end
+
+	def full?
+		self.cells.include?(" ") ? false : true
+	end
+
+	def turn_count
+		turns = 0
+		self.cells.each do |i|
```

Comment:
Use the highest level and most expressive iterator possible to solve a problem. [`count`](http://ruby-doc.org/core-2.2.0/Array.html#method-i-count) on the cells is perfect and the rest of this method is not needed. Entire body could read:

```ruby
def turn_count
  cells.count { |cell| cell == "X" || cell == "O" }
end
```